crates/wasmtime/build.rs

```diff
@@ -27,10 +27,10 @@ fn main() {
     custom_cfg("has_virtual_memory", has_virtual_memory);
     custom_cfg("has_host_compiler_backend", has_host_compiler_backend);
 
-    // If this OS isn't supported or if Cranelift doesn't support the host then
-    // there's no need to build these helpers.
+    // If this OS isn't supported or if Cranelift doesn't support the host or
+    // if no debug-builtins enabled then there's no need to build these helpers.
     #[cfg(feature = "runtime")]
-    if supported_os && has_host_compiler_backend {
+    if (supported_os && has_host_compiler_backend) || cfg!(feature = "debug-builtins") {
```

Comment:
Can this condition be massaged a bit to still include the `has_host_compiler_backend` conditional somehow? (or something like that)

Previously it was a nice property that compiling Wasmtime to an unknown/unsupported platform wouldn't involve compiling any C code and I'd ideally like to preserve that. An example situation where C code is not needed is a platform using Pulley even when debug-builtins is enabled because in such a situation the builtins aren't actually used by anything.